compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -2095,15 +2103,15 @@ trait Applications extends Compatibility {
       pt match
         case pt: FunProto =>
           if pt.applyKind == ApplyKind.Using then
-            val alts0 = alts.filterConserve(_.widen.stripPoly.isImplicitMethod)
+            val alts0 = alts.filterConserve(alt => widen(alt).stripPoly.isImplicitMethod)
```

Comment:
Eventually we might want to refactor this so the map lives on Context, and Type#widen uses that directly, so all calls to .widen downstream of resolveOverloaded benefit from the caching.  No urgency.

Reply:
I'm still worry about at which point we can safely cache the widen result. Since `resolveOverloaded` uses `constrainResult` and `normalizedCompatible`, which could constrain TypeVars, I guess the denotation may change as well?

Reply:
That's a very point. 👍🏼